File: webshop-frontend/src/app/model/part.ts
import {Brand} from './brand';

export class Part {

  id: string;
  public name: string;
  public description: string;
  public price: number;
  public priceOnSale: number;
  public onSale: boolean;
  public onStock: boolean;
  public stockNumber: number;
  public brand: Brand;

}

File: webshop-backend/src/main/java/com/webshop/model/Part.java
package com.webshop.model;

import lombok.Getter;
import lombok.Setter;

import javax.persistence.*;

/**
 * Created by user on 13.9.2018.
 */
@Table(name = "part")
@Entity
@Setter
@Getter

public class Part extends BaseEntity {

    private String name;

    private String description;

    private double price;

    private double priceOnSale;

    private boolean onSale;

    private boolean onStock;

    private int stockNumber;

    @ManyToOne(fetch = FetchType.EAGER)
    @JoinColumn(name = "brand_id")
    private Brand brand;
}
